contracts/ProductNft/ProductNft.sol

```diff
@@ -234,24 +234,33 @@ contract ProductNft is ERC1155, Ownable {
     }
 
     /**
-    *@dev   This function will switch the status of product deployment. 
-    *       If the token has not been deployed the status will be set to ready.
+    *@dev   This function will switch the status of product deployment to ready. 
+    *       If the token has not been deployed and the product status is 
+    *       not_ready the status will be set to ready.
     *       This can only be done for tokens of the Diamond/Rarer rarity.
-    *       Information regarding deployment will be retrieved using an API.
     *@param tokenIdList: the array of token Ids that is used to change the 
     *       deployment status of a token launched using the Valorize Token Launcher
-    *@param deployed: set to true if a token has been deployed 
     */
-    function switchProductStatus(uint256[] memory tokenIdList, bool deployed) public onlyOwner {
-        if (deployed) {
-            for(uint256 i=0; i < tokenIdList.length; i++) {
+    function switchProductStatusToReady(uint256[] memory tokenIdList) public onlyOwner {
+        for(uint256 i=0; i < tokenIdList.length; i++) {
+            require(tokenIdList[i] > startRarerTokenIdIndex && tokenIdList[i] < startRareTokenIdIndex, "Product status is already set to ready");
```

Comment:
I guess this message is not accurate, can we make the feedback for people calling this to be something like "Your token is not of the right type" or something descriptive like that